src/axolotl/monkeypatch/activation_offload_checkpoint.py

```
@@ -0,0 +1,250 @@
+        for i in range(len(outputs)):
+            if torch.is_tensor(outputs[i]) and outputs[i].requires_grad:
+                outputs_with_grad.append(outputs[i])
+                args_with_grad.append(args[i])
+        if len(outputs_with_grad) == 0:
+            raise RuntimeError("none of output has requires_grad=True")
+        torch.autograd.backward(outputs_with_grad, args_with_grad)
+        grads = tuple(
+            inp.grad if isinstance(inp, torch.Tensor) else None
+            for inp in detached_inputs
+        )
+        return (None, None) + grads
```

Ruff (0.15.15)

[warning] Consider `(None, None, *grads)` instead of concatenation

Replace with `(None, None, *grads)`

(RUF005)